packages/web-components/src/rating-display/rating-display.ts

```diff
@@ -0,0 +1,122 @@
+import { RatingDisplayColor, RatingDisplaySize } from './rating-display.options.js';
+
+/**
+ * The base class used for constructing a fluent-rating-display custom element
+ * @public
+ */
+export class RatingDisplay extends FASTElement {
+  /**
+   * The internal {@link https://developer.mozilla.org/docs/Web/API/ElementInternals | `ElementInternals`} instance for the component.
+   *
+   * @internal
+   */
+  public elementInternals: ElementInternals = this.attachInternals();
+
+  /**
+   * The color of the rating display icons.
+   *
+   * @public
+   * @default `marigold`
+   * @remarks
+   * HTML Attribute: `color`
+   */
+  @attr
+  public color: RatingDisplayColor = RatingDisplayColor.marigold;
+
+  /**
+   * Renders a single filled icon with a label next to it.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: `compact`
+   */
+  @attr({ attribute: 'compact', mode: 'boolean' })
+  public compact: boolean = false;
+  protected compactChanged(): void {
+    if (this.compact) {
+      this.max = 1;
+    }
+  }
+
+  /**
+   * The number of ratings.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: `count`
+   */
+  @attr({ converter: nullableNumberConverter })
+  public count?: number;
+
+  /**
+   * The maximum possible value of the rating.
+   * This attribute determines the number of icons displayed.
+   * Must be a whole number greater than 1.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: `max`
+   */
+  @attr({ converter: nullableNumberConverter })
+  public max: number = 5;
+
+  /**
+   * The size of the component.
+   *
+   * @public
+   * @default 'medium'
+   * @remarks
+   * HTML Attribute: `size`
+   */
+  @attr
+  public size: RatingDisplaySize = RatingDisplaySize.medium;
+
+  /**
+   * The value of the rating.
+   *
+   * @public
+   * @remarks
+   * HTML Attribute: `value`
+   */
+  @attr({ converter: nullableNumberConverter })
+  public value: number = 0;
+
+  private intlNumberFormatter = new Intl.NumberFormat();
+
+  public constructor() {
+    super();
+
+    this.elementInternals.role = 'img';
+  }
+
+  /**
+   * Returns "count" as string, formatted according to the locale.
+   *
+   * @internal
+   */
+  public get formattedCount(): string {
+    return this.count ? this.intlNumberFormatter.format(this.count) : '';
+  }
+
+  /**
+   * Generates the icon SVG elements based on the "max" attribute.
+   *
+   * @internal
+   */
+  public generateIcons(): string {
+    let htmlString: string = '';
+
+    // The value of the selected icon. Based on the "value" attribute, rounded to the nearest half.
+    const selectedValue: number = Math.round((this.compact ? 1 : this.value) * 2) / 2;
+
+    for (let i: number = 0; i < this.max * 2; i++) {
+      const iconValue: number = (i + 1) / 2;
+      htmlString += `<svg aria-hidden="true" ${
+        iconValue === selectedValue ? 'selected' : ''
+      }><use xlink:href="#star"></use></svg>`;
```

Comment:
Seems we no longer need to use `xlink:` namespace anymore: https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/xlink:href

Reply:
Good catch! I've removed `xlink:` and verified it still works as expected.